cosmwasm/utils.js

```diff
@@ -881,6 +886,55 @@ const CONTRACTS = {
     },
 };
 
+
+const downloadContractFromR2 = async (contractName, contractVersion) => {
+    const pathName = pascalToKebab(contractName);
+    const fileName = pascalToSnake(contractName);
+
+    const versionRegex = /^v\d+\.\d+\.\d+$/;
+    const commitRegex = /^[a-f0-9]{7,}$/;
+
+    let contractPath;
+
+    if (versionRegex.test(contractVersion)) {
+        // remove leading v
+        const semanticVersion = contractVersion.slice(1);      
+        contractPath = `${R2_BUCKET_URL}/releases/cosmwasm/${pathName}/${semanticVersion}/${fileName}.wasm`;
+    } else if (commitRegex.test(contractVersion)) {
+        contractPath = `${R2_BUCKET_URL}/pre-releases/cosmwasm/${contractVersion}/${fileName}.wasm`;
+    } else {
+        throw new Error(`Invalid contractVersion format: ${contractVersion}. Must be a semantic version (including prefix v) or a commit hash`);
+    }
+
+    console.log(`Downloading contract from ${contractPath}`);
+    const response = await fetch(contractPath);
+
+    if (!response.ok) {
+        throw new Error(`Failed to download contract: ${response.statusText}`);
+    }
+
+    // Ensure temp directory exists
+    const tempDir = path.join(__dirname, '../temp');
+
+    if (!existsSync(tempDir)) {
+        mkdirSync(tempDir, { recursive: true });
+    }
+
+    // Define local file path
+    const filePath = path.join(tempDir, `${fileName}.wasm`);
+
+    // Save the file locally
+    const fileStream = createWriteStream(filePath);
+    return new Promise((resolve, reject) => {
+        response.body.pipe(fileStream);
+        response.body.on('error', reject);
+        fileStream.on('finish', () => {
+            resolve(tempDir);
+        });
+    });
+};
```

Comment:
In general, keep a higher bar for code quality so it's more maintainable long term. It's easy to script something up quickly that works but harder to reason about later